clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -3205,6 +3205,40 @@ void CIRGenModule::applyReplacements() {
     auto NewF = dyn_cast<cir::FuncOp>(Replacement);
     assert(NewF && "not implemented");
 
+    // LLVM has opaque pointer but CIR not. So we may have to handle these
+    // different pointer types when performing replacement.
+    auto optionalUseRange = OldF.getSymbolUses(theModule);
+    if (optionalUseRange) {
+      for (auto U : *optionalUseRange) {
+        auto Call = mlir::dyn_cast<cir::CallOp>(U.getUser());
```

Comment:
What happens if the operation is a `TryCallOp`? Please add support for that too (or write a comment explaining why it only applies to regular `CallOp`).

Reply:
I add a comment to explain it will be much easier if we had something like `llvm::CallBase` which might be a base op interface `CallOp` and `TryCallOp`

Reply:
We have a `CIRCallOpInterface`... it's also totally within the scope to improve CIR in order to achieve your goals.